dot/state/offline_pruner.go

```diff
@@ -172,33 +169,48 @@ func (p *OfflinePruner) Prune() error {
 	}()
 
 	storagePrefixBytes := []byte(storagePrefix)
-	stream := inputDB.NewStream()
-	stream.ChooseKey = func(item *badger.Item) bool {
-		key := item.Key()
-		if !bytes.HasPrefix(key, storagePrefixBytes) {
-			// Ignore non-storage keys
-			return false
+
+	keyUpperBound := func(b []byte) []byte {
+		end := make([]byte, len(b))
+		copy(end, b)
+		for i := len(end) - 1; i >= 0; i-- {
+			end[i] = end[i] + 1
+			if end[i] != 0 {
+				return end[:i+1]
+			}
 		}
+		return nil // no upper-bound
+	}
+
+	prefixIterOptions := &pebble.IterOptions{
+		LowerBound: storagePrefixBytes,
+		UpperBound: keyUpperBound(storagePrefixBytes),
+	}
+
+	// Ignore non-storage keys
+	inputDBIter := inputDB.NewIter(prefixIterOptions)
```

Comment:
possible to reuse `pebble.NewPrefixIterator` here ?

Reply:
Yes, done!